app/src/main/java/com/framgia/fsalon/screen/scheduler/SchedulerViewModel.java

```diff
@@ -194,27 +205,87 @@ public void setLoadingMore(boolean loadingMore) {
+
     @Override
     public void onDateSet(DatePickerDialog view, int year, int monthOfYear, int dayOfMonth) {
-        Calendar cal = Calendar.getInstance();
-        cal.set(Calendar.YEAR, year);
-        cal.set(Calendar.MONTH, monthOfYear);
-        cal.set(Calendar.DAY_OF_MONTH, dayOfMonth);
-        cal.set(Calendar.HOUR_OF_DAY, 0);
-        cal.set(Calendar.MINUTE, 0);
-        cal.set(Calendar.SECOND, 0);
-        cal.set(Calendar.MILLISECOND, 0);
-        int time = (int) (cal.getTimeInMillis() / 1000);
-        mPresenter
-            .getSchedulers(FILTER_DAY, FIRST_PAGE, OUT_OF_INDEX, OUT_OF_INDEX, time, OUT_OF_INDEX);
-    }
-
-    @IntDef({TAB_TODAY, TAB_YESTERDAY, TAB_TOMORROW, TAB_CALENDAR})
+        if (mDatePickerDialog != null) mDatePickerDialog.dismiss();
+        mCalendar.set(Calendar.YEAR, year);
+        mCalendar.set(Calendar.MONTH, monthOfYear);
+        mCalendar.set(Calendar.DAY_OF_MONTH, dayOfMonth);
+        mCalendar.set(Calendar.HOUR_OF_DAY, 0);
+        mCalendar.set(Calendar.MINUTE, 0);
+        mCalendar.set(Calendar.SECOND, 0);
+        mCalendar.set(Calendar.MILLISECOND, 0);
+        String startDate = "", endDate = "";
+        switch (mTypeSelectDate) {
+            case TAB_START_DATE:
+                mStartDate = (int) (mCalendar.getTimeInMillis() / 1000);
+                mTypeSelectDate = TAB_END_DATE;
+                mDatePickerDialog = DatePickerDialog.newInstance(this, year,
+                    monthOfYear, dayOfMonth);
+                startDate = mCalendar.getTime().toString();
+                showDatePickerDialog();
+                return;
+            case TAB_SELECT_DATE:
+                setTitleTopSheet(mCalendar.getTime().toString());
+                mStartDate = (int) (mCalendar.getTimeInMillis() / 1000);
+                mPresenter
+                    .getSchedulers(FILTER_DAY, FIRST_PAGE, OUT_OF_INDEX, OUT_OF_INDEX, mStartDate,
+                        mEndDate);
+                mStartDate = OUT_OF_INDEX;
+                break;
+            case TAB_END_DATE:
+                mEndDate = (int) (mCalendar.getTimeInMillis() / 1000);
+                mPresenter
+                    .getSchedulers(FILTER_SPACE, FIRST_PAGE, OUT_OF_INDEX, OUT_OF_INDEX, mStartDate,
+                        mEndDate);
+                mEndDate = OUT_OF_INDEX;
+                mStartDate = OUT_OF_INDEX;
+                endDate = mCalendar.getTime().toString();
+                setTitleTopSheet("From " + startDate + "to " + endDate);
+            default:
+                break;
+        }
+    }
+
+    @IntDef({TAB_TODAY, TAB_YESTERDAY, TAB_TOMORROW, TAB_SPACE_TIME, TAB_SELECT_DATE})
     @Target(ElementType.PARAMETER)
     public @interface TabFilter {
         int TAB_TODAY = 0;
         int TAB_YESTERDAY = 1;
         int TAB_TOMORROW = 2;
-        int TAB_CALENDAR = 3;
+        int TAB_SPACE_TIME = 3;
+        int TAB_SELECT_DATE = 4;
+        int TAB_START_DATE = 5;
+        int TAB_END_DATE = 6;
```

Comment:
ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.